webhooks/mutating_admission_webhook.go

```diff
@@ -104,34 +104,37 @@ func CheckOwnerReference(pod *corev1.Pod, cm corev1.ConfigMap) bool {
 
 func (m *PodMutator) CreateConfigMap(ctx context.Context, name string, namespace string, pod *corev1.Pod) error {
 	m.Log.V(1).Info(fmt.Sprintf("Creating configmap %s", name))
-	reference := pod.OwnerReferences[0]
-	reference.Controller = m.falseVal()
+	references := []metav1.OwnerReference{
+		pod.OwnerReferences[0],
+	}
+	references[0].Controller = m.falseVal()
+	ff := m.GetFeatureFlag(ctx, name, namespace)
+	if ff.Name != "" {
+		references = append(references, m.GetFfReference(&ff))
+	}
 
-	spec := m.GetFeatureFlagSpec(ctx, name, namespace)
 	cm := corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      name,
 			Namespace: namespace,
 			Annotations: map[string]string{
 				"openfeature.dev/featureflagconfiguration": name,
 			},
-			OwnerReferences: []metav1.OwnerReference{
-				reference,
-			},
+			OwnerReferences: references,
 		},
 		Data: map[string]string{
-			"config.yaml": spec.FeatureFlagSpec,
+			"config.yaml": ff.Spec.FeatureFlagSpec,
 		},
 	}
 	return m.Client.Create(ctx, &cm)
 }
 
-func (m *PodMutator) GetFeatureFlagSpec(ctx context.Context, name string, namespace string) configv1alpha1.FeatureFlagConfigurationSpec {
+func (m *PodMutator) GetFeatureFlag(ctx context.Context, name string, namespace string) configv1alpha1.FeatureFlagConfiguration {
 	ffConfig := configv1alpha1.FeatureFlagConfiguration{}
 	if err := m.Client.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, &ffConfig); errors.IsNotFound(err) {
-		return configv1alpha1.FeatureFlagConfigurationSpec{}
+		return configv1alpha1.FeatureFlagConfiguration{}
 	}
-	return ffConfig.Spec
+	return ffConfig
 }
 
 func (m *PodMutator) InjectSidecar(pod *corev1.Pod, configMap string) ([]byte, error) {
```

Comment:
like many of the methods on this object, this can be private

Reply:
done